litr/src/main/cpp/audio-resampler.cpp

```diff
@@ -0,0 +1,77 @@
+#include <jni.h>
+#include "oboe_resampler/MultiChannelResampler.h"
+
+using namespace resampler;
+
+MultiChannelResampler* audio_resampler = nullptr;
+int channel_count = -1;
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_linkedin_android_litr_render_AudioRenderer_initAudioResampler(
+        JNIEnv* env,
+        jobject /* this */,
+        jint channelCount,
+        jint sourceSampleRate,
+        jint targetSampleRate) {
+    audio_resampler = MultiChannelResampler::make(
+            channelCount,
+            sourceSampleRate,
+            targetSampleRate,
+            MultiChannelResampler::Quality::Fastest);
+    channel_count = channelCount;
+}
+
+extern "C" JNIEXPORT int JNICALL
+Java_com_linkedin_android_litr_render_AudioRenderer_resample(
+        JNIEnv* env,
+        jobject,
+        jobject jsourceBuffer,
+        jint sampleCount,
+        jshortArray jtargetBuffer) {
+    if (audio_resampler != nullptr && channel_count > 0) {
+        auto sourceBuffer = (jbyte *) env->GetDirectBufferAddress(jsourceBuffer);
+        jshort* targetBuffer = env->GetShortArrayElements(jtargetBuffer, JNI_FALSE);
+
+        auto inputBuffer = new float[sampleCount * channel_count];
+        auto outputBuffer = new float[channel_count];
+
+        // bytes contained in audio buffer produced by MediaCodec contains make up big endian shorts
+        // first we recreate short values, then simply cast them to floats, expected by Oboe resampler
+        for (int index = 0; index < sampleCount * channel_count; index++) {
+            inputBuffer[index] = (float) ((short) ((sourceBuffer[index * 2] << 8) | sourceBuffer[index * 2 + 1]));
```

Comment:
jbytes are signed. So the bytes will get sign extended, which corrupts the higher bytes.
Either mask the signed bytes with 0xFF or make the sourceBuffer a (uint8_t *).